pkg_resources/__init__.py

```diff
@@ -804,9 +806,14 @@ def resolve(self, requirements, env=None, installer=None,
                 continue
             # If the req has a marker, evaluate it -- skipping the req if
             # it evaluates to False.
-            # https://github.com/pypa/setuptools/issues/523
-            _issue_523_bypass = True
-            if not _issue_523_bypass and req.marker and not req.marker.evaluate():
+            if req.marker:
+                result = []
+                if req in extra_req_mapping:
+                    for extra in extra_req_mapping[req] or ['']:
```

Comment:
@s-t-e-v-e-n-k, @rbtcollins What is the `or ['']` for here? The tests pass without it. Can I remove it? If not, can you create a test to capture the use case that drives it?